compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -624,3 +621,85 @@ pub fn create_compressed_metadata_file_for_xcoff(
     file.append_section_data(section, data, 1);
     file.write().unwrap()
 }
+
+/// Creates a simple WebAssembly object file, which is itself a wasm module,
+/// that contains a custom section of the name `section_name` with contents
+/// `data`.
+///
+/// NB: the wasm file format is simple enough that for now an extra crate from
```

Comment:
Do you anticipate us needing to pull in these crates eventually? I guess the file format itself is stable so what we're doing currently should continue working but perhaps we'll want to take advantage of other/newer features for various reasons?

Reply:
Unsure! Had you asked me a year or two ago I would have thought that even this wouldn't be needed, so I'm not certain what the future has in store in that regard. My hope though is that if it's necessary to pull in the crates that it should be pretty easy to do so. I mostly didn't want to add any new dependencies just yet while it wasn't necessary